### DISCUSSION TITLE
"Error: No matching key" logged in the wallet

### DISCUSSION THREAD

**Discussion by fanboyn** [WalletConnect/walletconnect-monorepo#3702] [Developer Support]
Hey,

We're struggling with an error happening from time to time. We've updated to the latest release of @walletconnect libs (2.9.1) and are still seeing it:

{ context: 'client' }, [Error: No matching key. session: 9a05b6099c6cc662ed39dd43795adec83ad64dfc6cfd7779ae1bc3d6470bb5c6]

This is happening in [Valora](https://github.com/valora-inc/wallet), in a React Native environment (0.70.12).
We don't yet have reproduction steps. But we've seen it when testing with https://curve.fi/ so far.

The symptom is the user does an action in a dapp that is already connected to Valora and Valora doesn't show the modal to approve it. But instead logs the error above.

Connecting to new dapps still work after that, but any action from the dapp doesn't show the modal in Valora and logs the error.

We've seen that clearing local storage allows fixing the issue. But for real users, the only option to recover from this is to reinstall Valora and re-import their account. Which is a big problem.

For our users, it looks like Valora is broken because it doesn't show the modal to approve the action.

We've searched through the WalletConnect repo and found many related discussions:
https://github.com/search?q=org%3AWalletConnect+Error%3A+%22No+matching+key.%22&type=discussions

But none so far giving a definitive way to fix it for good.

Do you have any recommendation on what to do next?
Let us know if you need more information.

**Comment by MuckT** [comment]
@glitch-txs it looks like these users are reposting previously answered questions and marking themselves as answers, you can see they've used pre-existing text from previous issues. Maybe they're just new, but it's certainly weird.

~~[This comment](https://github.com/orgs/WalletConnect/discussions/3702#discussioncomment-7089178) from @horaciomota is a direct copy of my earlier comment [here](https://github.com/orgs/WalletConnect/discussions/3291#discussioncomment-6700806)~~ Comment was deleted.

This post is a copy of [this existing post](https://github.com/orgs/WalletConnect/discussions/3291#discussion-5446353).

Neither of these developers are developers at Valora.

The same can be seen for the issue: https://github.com/orgs/WalletConnect/discussions/3701 is a copy of https://github.com/orgs/WalletConnect/discussions/3683.
